streamer/src/nonblocking/quic.rs

```diff
@@ -85,6 +89,58 @@ struct PacketAccumulator {
     pub chunks: Vec<PacketChunk>,
 }
 
+struct StakedStreamLoadEMA {
+    current_load_ema: AtomicU64,
+    load_in_recent_interval: Arc<AtomicU64>,
+}
+
+impl StakedStreamLoadEMA {
+    fn new() -> Self {
+        Self {
+            current_load_ema: AtomicU64::default(),
+            load_in_recent_interval: Arc::new(AtomicU64::default()),
+        }
+    }
+
+    async fn update_ema(self: Arc<Self>, exit: Arc<AtomicBool>, stats: Arc<StreamStats>) {
+        while !exit.load(Ordering::Relaxed) {
```

Comment:
Practically it should be fine. I thought of doing that initially.
Only catch is if there were no streams opened for some duration, the EMA will become incorrect/stale. Not sure if there's a possibility of that, or how bad will it be if it were to go stale.

Reply:
i think the staleness issue is resolved by parameterizing the ema calculation on time

Reply:
@t-nelson we'll need to wrap `last_update` in `RwLock`, since it can be updated from different async tasks (connection handlers). Making sure we are happy with that before making the change.

Reply:
Updated the code to remove async task